File: MAP/Lab 2/ToyInterpreter/src/view/commands/RunOneStepCommand.java
package view.commands;

import controller.Controller;

import java.util.Scanner;

public class RunOneStepCommand extends Command {

    public RunOneStepCommand(String key, String description, Controller controller) {
        super(key, description, controller);
    }

    @Override
    public void execute() throws Exception {
        int index = getNr("index: ");
        executeOneStepAt(index);
        printProgramAt(index);
    }

    private int getNr(String msg) {
        System.out.println(msg);

        Scanner sc = new Scanner(System.in);
        return sc.nextInt();
    }

    private void executeOneStepAt(int index) throws Exception {
        controller.oneStep( controller.getProgramAt(index) );
    }

    private void printProgramAt(int index) {
        System.out.println( controller.getProgramAt(index).toString()  );
    }
}

File: MAP/Lab10/ConsoleApp1/ConsoleApp1/View/commands/RunOneStepCommand.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToyInterpreter.View.commands
{
    class RunOneStepCommand: Command
    {
        public RunOneStepCommand(String key, String description, Controller.Controller controller): base(key, description, controller)
        {
        }


        public override void execute()
        {
            executeOneStep();
            printProgram();
        }

        private int getNr(String msg)
        {
            Console.WriteLine(msg);
  
            return Int32.Parse(Console.ReadLine());
        }

        private void executeOneStep()
        {
            controller.oneStep(controller.getFirstProgram());
        }

        private void printProgram()
        {
            Console.WriteLine(controller.getFirstProgram());
        }
    }
}
